### PR TITLE
docs: use relative links for development docs

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Replaces hardcoded links with relative links where possible
- Improve link names
- Remove `.` at the end of list items

## Context

Helps with #13400, but doesn't close it.

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: docs/development/new-package-manager-template.md [hunk 1]
```diff
@@ -23,7 +23,7 @@
 
 ### What type of package files and names does it use?
 
-### What [fileMatch](https://renovatebot.com/docs/configuration-options/#filematch) pattern(s) should be used?
+### What [fileMatch](../usage/configuration-options.md/#filematch) pattern(s) should be used?
```
Comment (HonkingGoose) [inline docs/development/new-package-manager-template.md:26]: This link breaks in VS code, but works on GitHub.

File: docs/development/issue-labeling.md [hunk 2]
```diff
@@ -18,10 +18,10 @@ These labels should also map approximately to our Conventional Commit scopes.
...
 You should know about platforms, package managers, datasources and versioning to label issues effectively.
 
-- To learn about platforms, read the [Renovate docs on Platforms](https://docs.renovatebot.com/modules/platform/).
-- To learn about managers, read the [Renovate docs on Managers](https://docs.renovatebot.com/modules/manager/).
-- To learn about datasources, read the [Renovate docs on Datasources](https://docs.renovatebot.com/modules/datasource/).
-- To learn more about versioning, read the [Renovate docs on Versioning](https://docs.renovatebot.com/modules/versioning/).
+- To learn about platforms, read the [Renovate docs on Platforms](../usage/modules/platform.md)
+- To learn about managers, read the [Renovate docs on Managers](../usage//modules/manager.md)
+- To learn about datasources, read the [Renovate docs on Datasources](../usage/modules/datasource.md)
+- To learn more about versioning, read the [Renovate docs on Versioning](../usage/modules/versioning.md)
```
Comment (HonkingGoose) [inline docs/development/issue-labeling.md:24]: Maybe I should revert this, as the published docs have all the information, the files in the repository are missing the "autogenerated" parts.

File: docs/development/configuration.md [hunk 3]
```diff
@@ -66,6 +66,6 @@ If you add configuration options to your `package.json` then these will override
 
 ## Configuration Options
 
-Please see [https://docs.renovatebot.com/configuration-options/](https://docs.renovatebot.com/configuration-options/) for a list of user-facing configuration options.
+Please read [our list of user-facing configuration options](../usage/configuration-options.md).
```
Comment (viceice) [inline docs/development/configuration.md:69]: those docs links should probably use the live docs, as the repo files are missing a lot generated content

File: docs/development/issue-labeling.md (hunk 2 above)
Comment (HonkingGoose) [reply docs/development/issue-labeling.md:24]:
```suggestion
- To learn about platforms, read the [Renovate docs on Platforms](https://docs.renovatebot.com/modules/platform/)
- To learn about managers, read the [Renovate docs on Managers](https://docs.renovatebot.com/modules/manager/)
- To learn about datasources, read the [Renovate docs on Datasources](https://docs.renovatebot.com/modules/datasource/)
- To learn more about versioning, read the [Renovate docs on Versioning](https://docs.renovatebot.com/modules/versioning/)
```

File: docs/development/configuration.md (hunk 3 above)
Comment (HonkingGoose) [reply docs/development/configuration.md:69]:
```suggestion
Please read [our list of user-facing configuration options](https://docs.renovatebot.com/configuration-options/).
```

**Review by rarkins — APPROVED** [review]

**Pull request merged**